ntbs-service/Jobs/StoredProcedureJobBase.cs

```diff
@@ -1,94 +1,31 @@
     {
-        protected readonly IConfiguration _configuration;
-        protected readonly string _connectionString;
-        protected string _sqlString = "";
-        protected object[] _parameters;
-        protected PerformContext _context;
-
-        public StoredProcedureJobBase(IConfiguration configuration, string sqlString, object[] sqlParameters = null)
-        {
-            _configuration = configuration;
-            _connectionString = _configuration.GetConnectionString(Constants.DbConnectionStringReporting);
-            _sqlString = sqlString;
-            _parameters = sqlParameters;
-        }
-
-        public StoredProcedureJobBase(IConfiguration configuration)
-        {
-            _configuration = configuration;
-            _connectionString = _configuration.GetConnectionString(Constants.DbConnectionStringReporting);
-        }
-
-        /// PerformContext context is passed in via Hangfire Server
-        public virtual async Task Run(PerformContext context, IJobCancellationToken token)
-        {
-            _context = context;
-
-            Log.Information($"Starting stored procedure job");
-            _context.WriteLine($"Starting stored procedure job");
-
-            if (string.IsNullOrEmpty(_sqlString))
-            {
-                throw new ArgumentNullException(nameof(_sqlString));
-            }
-
-            var successfulExecution = false;
-            var resultChanges = 0;
-
-            try
-            {
-                var result = await ExecuteStoredProcedure(token);
-                resultChanges = result.Count();
-                successfulExecution = DidExecuteSuccessfully(result);
-            }
-            finally
-            {
-                Log.Information($"Finishing stored procedure job with {resultChanges} changes made, successful? {successfulExecution}.");
-                _context.WriteLine($"Finishing stored procedure job with {resultChanges} changes made, successful? {successfulExecution}.");
-            }
-        }
-
-        protected virtual async Task<IEnumerable<dynamic>> ExecuteStoredProcedure(IJobCancellationToken token)
+        protected static void CheckExecutedSuccessfully(PerformContext context, IEnumerable<dynamic> resultToTest)
         {
-            IEnumerable<dynamic> result = new List<dynamic>();
+            var serialisedResult = JsonConvert.SerializeObject(resultToTest);
+            LogInfo(context, $"Result: {serialisedResult}");
 
-            using (var connection = new SqlConnection(_connectionString))
+            if (resultToTest.Any())
             {
-                connection.Open();
-                result = await connection.QueryAsync(_sqlString, _parameters, null, Constants.SqlServerDefaultCommandTimeOut, System.Data.CommandType.StoredProcedure);
+                throw new ApplicationException(
+                    "Stored procedure did not execute successfully as result has messages, check the logs");
```

Comment:
Checking the logs can be a bit of a pain - should we include the result in the error message here? Or are we worried that it could contain sensitive information?